rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.template

```diff
@@ -182,7 +182,7 @@ convert_ros_to_dds(const void * untyped_ros_message, void * untyped_dds_message)
       return false;
     }
     DDS_Long length = static_cast<DDS_Long>(size);
-    dds_message->@(field.name)_.length(length);
+    dds_message->@(field.name)_.ensure_length(length, 2 * length);
```

Comment:
That's the max, it only allocates if there is not enough memory of length. I suppose, instead, it should be something like `current_capacity * 2`.